src/common/utxobased/engine/ServerStates.ts

```diff
@@ -304,38 +320,117 @@ export function makeServerStates(config: ServerStateConfig): ServerStates {
 
   const broadcastTx = async (transaction: EdgeTransaction): Promise<string> => {
     return await new Promise((resolve, reject) => {
```

Comment:
Since you are refactoring this guy pretty heavily, could we fix the strange `new Promise` thing into normal async / await? The structure seems pretty linear, so it should be a straightforward conversion.

OTOH, what you have works, even if it weird to read.

Reply:
I was going to, but it would change the behavior from parallel promises to sequential which is not what is expected for broadcast. I'm just honoring the implementation; but if you have a good argument for sequentializing it then I can consider.

Reply:
Right, which is why we normally use `Promise.all` in cases like this. It's fine to leave it as-is, though.